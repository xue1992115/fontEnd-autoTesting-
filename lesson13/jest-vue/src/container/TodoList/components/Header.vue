<template>
    <div class="header">
      <div class="header-content">
        TodoList: <input data-test="input" v-model="inputValue" @keyup.enter="changeInput" class="input" placeholder="请添加todo itemw">
      </div>
    </div>
</template>

<script>
export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Header',
  data () {
    return {
      inputValue: '123'
    }
  },
  methods: {
    changeInput () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>

<style lang="stylus" scoped>
.header {
  background: #666
  line-height: 60px
  &-content {
    width: 600px
    margin: 0 auto
    color: #333
    font-size: 24px
    .input {
      float: right
      width: 300px
      line-height: 24px
      outline: none
      margin-top: 16px
      color: #333
      font-size: 30px
    }
  }
}
</style>
