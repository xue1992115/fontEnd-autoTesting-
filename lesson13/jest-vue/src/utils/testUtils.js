export const findTestWrapper = (wrapper, tag) => wrapper.find(tag)
