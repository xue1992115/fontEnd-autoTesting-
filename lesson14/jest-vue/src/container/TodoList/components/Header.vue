<template>
    <div class="header">
      <div class="header-content">
        TodoList: <input data-test="header-input" :value="inputValue" @keyup.enter="addItem" class="input" placeholder="请添加todo item"
        @input="(e) => changeInputValue(e.target.value)">
      </div>
    </div>
</template>

<script>
import { mapState, mapMutations } from 'vuex'
export default {
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Header',
  // data () {
  //   return {
  //     inputValue: '123'
  //   }
  // },
  computed: {
    ...mapState(['inputValue'])
  },
  methods: {
    ...mapMutations({
      changeInputValue: 'changeInputValue',
      clearInput: 'clearInput'
    }),
    addItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.clearInput()
      }
    }
  }
}
</script>

<style lang="stylus" scoped>
.header {
  background: #666
  line-height: 60px
  &-content {
    width: 600px
    margin: 0 auto
    color: #333
    font-size: 24px
    .input {
      float: right
      width: 300px
      line-height: 24px
      outline: none
      margin-top: 16px
      color: #333
      font-size: 30px
    }
  }
}
</style>
