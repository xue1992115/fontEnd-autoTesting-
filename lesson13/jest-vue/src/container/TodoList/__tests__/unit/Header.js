/* eslint-disable no-undef */
// vue提供的测试工具
import { shallowMount } from '@vue/test-utils'
import Header from '../../components/Header.vue'
import { findTestWrapper } from '../../../../utils/testUtils'
// 使用快照功能，就是当Header组件样式或者功能放生改变的时候，做一个提醒
it('Header 样式发生改变，做提示', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot()
})
// 单元测试用例
it('Header 包含input框', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, '[data-test="input"]')
  // wrapper.exists api表示是否存在这个节点
  expect(input.exists()).toBe(true)
})
// 也可以不添加describe
it('Header 包含input框 初始值为空的', () => {
  const wrapper = shallowMount(Header)
  const inputValue = wrapper.vm.$data.inputValue
  // wrapper.exists api表示是否存在这个节点
  expect(inputValue).toBe('123')
})

// 测试修改input框中的值
it('Header 中input框值发生变化，数据应该跟着变化', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, '[data-test="input"]')
  input.setValue('hxx')
  const inputValue = wrapper.vm.$data.inputValue
  // wrapper.exists api表示是否存在这个节点
  expect(inputValue).toBe('hxx')
})
// 当用户什么都没有输入时候，点击回车按键
// 希望不触发add事件
it('Header中input框输入回车无内容时，无反应', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, '[data-test="input"]')
  input.setValue('')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeFalsy()
})
// 当用户输入内容，点击回车按键， 触发事件
it('Header中input框输入回车有内容时，触发事件', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, '[data-test="input"]')
  input.setValue('hxx')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
})
// 当用户输入内容，点击回车按键， 触发事件并且清空内容
it('Header中input框输入回车有内容时，触发事件， 同时清空inputValue', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper, '[data-test="input"]')
  input.setValue('hxx')
  input.trigger('keyup.enter')
  expect(wrapper.emitted().add).toBeTruthy()
  expect(wrapper.vm.$data.inputValue).toBe('')
})
