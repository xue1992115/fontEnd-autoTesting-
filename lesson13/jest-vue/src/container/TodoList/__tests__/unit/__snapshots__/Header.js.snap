// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，做提示 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="header"
      >
        <div
          class="header-content"
        >
           TodoList: 
          <input
            class="input"
            data-test="input"
            placeholder="请添加todo itemw"
          />
        </div>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {
        "VTU_COMPONENT": Object {
          "changeInput": [Function],
          "inputValue": "123",
        },
      },
      "da": null,
      "data": Object {},
      "devtoolsRawSetupState": Object {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": Object {
            "changeInput": [Function],
            "inputValue": "123",
          },
          "_value": Object {
            "changeInput": [Function],
            "inputValue": "123",
          },
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {
        "VTU_COMPONENT": Object {
          "changeInput": [Function],
          "inputValue": "123",
        },
      },
      "refs": Object {
        "VTU_COMPONENT": Object {
          "changeInput": [Function],
          "inputValue": "123",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "detached": true,
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": Object {
        "VTU_COMPONENT": Object {
          "changeInput": [Function],
          "inputValue": "123",
        },
      },
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "__v_isReadonly": 0,
            "__v_isRef": 0,
            "__v_isShallow": 0,
            "__v_raw": 0,
            "__v_skip": 4,
            "_isMockFunction": 0,
            "changeInput": 4,
            "constructor": 0,
            "inputValue": 2,
            "isVueInstance": 0,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "changeInput": [Function],
            "inputValue": "123",
          },
          "da": null,
          "data": Object {
            "inputValue": "123",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {},
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {},
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "changeInput": [Function],
            "inputValue": "123",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "detached": true,
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": Array [
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " TodoList: ",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el":  TodoList: ,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                  Object {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": null,
                    "component": null,
                    "ctx": [Circular],
                    "dirs": Array [
                      Object {
                        "arg": undefined,
                        "dir": Object {
                          "beforeUpdate": [Function],
                          "created": [Function],
                          "mounted": [Function],
                        },
                        "instance": Object {
                          "changeInput": [Function],
                          "inputValue": "123",
                        },
                        "modifiers": Object {},
                        "oldValue": undefined,
                        "value": "123",
                      },
                    ],
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <input
                      class="input"
                      data-test="input"
                      placeholder="请添加todo itemw"
                    />,
                    "key": null,
                    "patchFlag": 544,
                    "props": Object {
                      "class": "input",
                      "data-test": "input",
                      "onKeyup": [Function],
                      "onUpdate:modelValue": [Function],
                      "placeholder": "请添加todo itemw",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "input",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="header-content"
                >
                   TodoList: 
                  <input
                    class="input"
                    data-test="input"
                    placeholder="请添加todo itemw"
                  />
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": Object {
                  "class": "header-content",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "ctx": [Circular],
                "dirs": Array [
                  Object {
                    "arg": undefined,
                    "dir": Object {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": Object {
                      "changeInput": [Function],
                      "inputValue": "123",
                    },
                    "modifiers": Object {},
                    "oldValue": undefined,
                    "value": "123",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  class="input"
                  data-test="input"
                  placeholder="请添加todo itemw"
                />,
                "key": null,
                "patchFlag": 544,
                "props": Object {
                  "class": "input",
                  "data-test": "input",
                  "onKeyup": [Function],
                  "onUpdate:modelValue": [Function],
                  "placeholder": "请添加todo itemw",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="header"
            >
              <div
                class="header-content"
              >
                 TodoList: 
                <input
                  class="input"
                  data-test="input"
                  placeholder="请添加todo itemw"
                />
              </div>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": Object {
              "class": "header",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "data": [Function],
            "methods": Object {
              "changeInput": [Function],
            },
            "name": "Header",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="header"
        >
          <div
            class="header-content"
          >
             TodoList: 
            <input
              class="input"
              data-test="input"
              placeholder="请添加todo itemw"
            />
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "ref": "VTU_COMPONENT",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "data": [Function],
          "methods": Object {
            "changeInput": [Function],
          },
          "name": "Header",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="header"
        >
          <div
            class="header-content"
          >
             TodoList: 
            <input
              class="input"
              data-test="input"
              placeholder="请添加todo itemw"
            />
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.45",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": Array [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": Object {
    "changeInput": [Function],
    "inputValue": "123",
  },
  "isDisabled": [Function],
  "rootVM": Object {
    "VTU_COMPONENT": Object {
      "changeInput": [Function],
      "inputValue": "123",
    },
  },
  "wrapperElement": <div
    class="header"
  >
    <div
      class="header-content"
    >
       TodoList: 
      <input
        class="input"
        data-test="input"
        placeholder="请添加todo itemw"
      />
    </div>
  </div>,
}
`;
